src/app.rs

```diff
@@ -264,6 +264,17 @@ impl Config {
     pub fn get_repoconfig(&self, name: &str) -> Result<&RepoConfig, ApiError> {
         self.repos.get(name).ok_or_else (|| ApiError::BadRequest("No such repo".to_string()))
     }
+
+    pub fn get_repoconfig_from_path(&self, path: &Path) -> Result<&RepoConfig, ApiError> {
+        for ancestor in path.ancestors() {
+            if let Some(name) = ancestor.to_str() {
+                if let Some(config) = self.repos.get(name) {
+                    return Ok(config)
+                }
+            }
+        }
+        Err(ApiError::BadRequest("No such repo".to_string()))
```

Comment:
I don't like this implementation, its O(n*m) where n = depth of passed in path, m = number of repos. The problem is that path.ancestors() always starts at the depthmost.

Can you instead just loop over all the repos and see if path.starts_with(repo.name) instead?

Reply:
Fair. I'll see what I can come up with.